### PR TITLE
DMP-700_Implement_REST_endpoint_for_get_case_by_ID

### PR DESCRIPTION
**Pull request by marcmollins**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-700

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/resources/openapi/cases.yaml [hunk 1]
```diff
@@ -76,6 +76,31 @@ paths:
         '500':
           description: Internal Server Error
 
+  /cases/{caseId}:
+    get:
+      tags:
+        - Cases
+      summary: Allows the retrieval of a single case using a given caseId
+      description: |-
+        Retrieves a single hearing for a single caseId
```
Comment (cmagowan-hmcts) [inline src/main/resources/openapi/cases.yaml:85]: Can you fix this description?

File: src/main/resources/openapi/cases.yaml [hunk 2]
```diff
@@ -224,6 +249,9 @@ components:
       required:
         - courthouse
       properties:
+        case_id:
```
Comment (cmagowan-hmcts) [inline src/main/resources/openapi/cases.yaml:252]: we should maybe remove this again, if we are not going to use it.

File: src/main/resources/openapi/cases.yaml [hunk 3]
```diff
@@ -260,6 +288,45 @@ components:
           type: array
           items:
             $ref: '#/components/schemas/defence'
+        retain_until:
```
Comment (cmagowan-hmcts) [inline src/main/resources/openapi/cases.yaml:291]: we should maybe remove this again, if we are not going to use it.

File: src/main/java/uk/gov/hmcts/darts/cases/controller/CaseController.java [hunk 4]
```diff
@@ -109,4 +110,13 @@ public ResponseEntity<List<Hearing>> casesCaseIdHearingsGet(
         return new ResponseEntity<>(caseService.getCaseHearings(caseId), HttpStatus.OK);
     }
 
+    @Override
+    public ResponseEntity<SingleCase> casesCaseIdGet(
+        @Parameter(name = "caseId", description = "caseId is the internal cas_id of the case.", required = true, in = ParameterIn.PATH) @PathVariable("caseId") Integer caseId
```
Comment (chrisbellingham-hmcts) [inline src/main/java/uk/gov/hmcts/darts/cases/controller/CaseController.java:115]: Do we need to repeat these annotations/constraints from the interface, will they not be inherited?

File: src/main/resources/openapi/cases.yaml [hunk 5]
```diff
@@ -260,6 +288,45 @@ components:
...
+        courthouse:
+          type: string
+          example: SWANSEA
+        case_number:
+          type: string
+          example: CASE1001
+        upload_priority:
+          type: string
+          example: 10
+        defendantList:
```
[the remark this answers is not in the document]
Comment (cmagowan-hmcts) [reply src/main/resources/openapi/cases.yaml:311]: we have a slight problem with that, as defence cannot easily be pluralised, so it was decided to stick 'List' on the end of everything to make it consistent.

File: src/main/java/uk/gov/hmcts/darts/cases/controller/CaseController.java (hunk 4 above)
Comment (marcmollins) [reply src/main/java/uk/gov/hmcts/darts/cases/controller/CaseController.java:115]: yes, I believe so, but all other methods have these annotations - so should I update each method and remove the annotations?

**Review by cmagowan-hmcts — APPROVED** [review]

**Pull request merged**